js/src/ui/DappVouchFor/store.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { action, observable } from 'mobx';
+import { uniq } from 'lodash';
+
+import Contracts from '~/contracts';
+import { vouchfor as vouchForAbi } from '~/contracts/abi';
+
+export default class Store {
+  @observable vouchers = [];
+
+  constructor (api, app) {
+    this._api = api;
+    this._app = app;
+
+    Contracts
```

Comment:
That won't really work in case of re-connection or dynamic change, but I supposed we don't need that for initial PoC.

Reply:
100%, added to list of next PR cleanups.